ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -217,6 +230,34 @@ public CancelHandle receiveInvocationRequest(@NotNull final InvocationRequest in
         return cancellationFlag::cancel;
     }
 
+    private static <A> boolean intersect(final Set<A> fst, final Set<A> snd) {
+        fst.retainAll(snd);
+        return !fst.isEmpty();
+    }
+
+    private static Set<String> typesOf(final Object o) {
```

Comment:
In this case it would be far more efficient to simply iterate a list of ``Class<?>`` and do a ``clazz.isInstance()`` on each.  But this is also non-ideal.  As I said in chat, interceptors adding non-serializable items should be removing them again on exit.